src/font-stylesheet-timeout.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isDocumentReady} from './document-ready';
+import {timerFor} from './timer';
+
+/**
+ * While browsers put a timeout on font downloads (3s by default,
+ * some less on slow connections), there is no such timeout for style
+ * sheets. In the case of AMP external stylesheets are ONLY used to
+ * download fonts, but browsers have no reasonable timeout for
+ * stylesheets. Users may thus wait a long time for these to download
+ * even though all they do is reference fonts.
+ *
+ * For that reasons this function identifies fonts that have not
+ * downloaded within 1 second of the page response starting and turns
+ * the respective reinserts the link tags dynamically. This removes
+ * their blocking nature and lets the doc render.
+ *
+ * 1 second was picked, because the fonf-stylesheets are typically
+ * tiny. If a connection wasn't able to deliver them within 1s
+ * of page load start, then it is unlikely that it will be able
+ * to download the font itself within 3s.
+ *
+ * @param {!Window} win
+ */
+export function fontStylesheetTimeout(win) {
+  let timeSinceResponseStart = 0;
+  // If available, we start counting from the time the HTTP response
+  // for the page started. The preload scanner should then quickly
+  // start the CSS download.
+  const perf = win.performance;
+  if (perf && perf.timing && perf.timing.responseStart) {
+    timeSinceResponseStart = Date.now() - perf.timing.responseStart;
+  }
+  const timeout = Math.max(1, 1000 - timeSinceResponseStart);
+
+  timerFor(win).delay(() => {
+    // We waited for the timeout period. There is no way to check whether
+    // the stylesheets actually loaded. For that reason we check whether
```

Comment:
I don't agree that there's no way. Y'all could require authors to include an `onload` attribute that invokes a handler and tags loaded stylesheet `link`s. See [this minimal example](http://jsbin.com/jequwatofa/edit?html,output). This may be a bad call performance-wise for reasons not clear to me rn.

More importantly, it's clunky and y'all may not want to ask authors to do it. Besides, maybe just reinserting all the `link`s in the doc works just fine.